GsaGH/Parameters/1_Properties/GsaMaterial.cs

```diff
@@ -479,6 +504,116 @@ public enum MatType {
       }
     }
 
+    private void ValidateApiMaterial() {
+      List<string> gradeNames = GetGradeNames(
+        MaterialType, SteelDesignCodeName, ConcreteDesignCodeName);
+      if (gradeNames.Contains(Name)) {
+        AnalysisMaterial a = CreateCodeAnalysisMaterial(Name);
+        if (AnalysisMaterialsAreEqual(a, AnalysisMaterial)) {
+          return;
+        }
+
+        _gradeName = Name;
+      } else {
+        var dict = new Dictionary<string, double>();
+        foreach (string gradeName in gradeNames) {
+          AnalysisMaterial a = CreateCodeAnalysisMaterial(gradeName);
+          if (AnalysisMaterialsAreEqual(a, AnalysisMaterial)) {
+            _gradeName = gradeName;
+            return;
+          }
+
+          dict[gradeName] = AnalysisMaterialsSimilarity(a, AnalysisMaterial);
+        }
+
+        _gradeName = dict.Aggregate((x, y) => x.Value > y.Value ? x : y).Key;
+      }
+    }
+
+    private double AnalysisMaterialsSimilarity(AnalysisMaterial a, AnalysisMaterial b) {
+      if (a == null || b == null) {
+        return 0;
+      }
+
+      double similarity = 0;
+      if (Math.Round(a.CoefficientOfThermalExpansion, 11) ==
```

Comment:
can you change it pls to condition ? true : false schema?

Reply:
removed this